Enterprise-CMCS/macpro-onemac
# OY2 22143: Withdraw Formal RAI Response action - State

Story: https://qmacbis.atlassian.net/browse/OY2-22143
Endpoint: See github-actions bot comment

### Details
State users should be able to submit a request to withdraw the most recent Formal RAI Response.

### Changes
- added two new statuses: "RAI Response Withdraw Enabled" and “Formal RAI Response - Withdrawal Requested”
- added the "Withdraw Formal RAI Response" action to state view dashboard and details page
- updated seed data to include test packages in the status needed for the story.
- created the "Withdraw Formal RAI Response” form
- updated the other withdraw types because code assumed that if there were no required documents, then it was an either/or requirement between attachments and additional information, but this form requires the additional information with optional attachments, so a new requirements pattern.
- updated the FileUploader component to set ready callback to true if there are no required attachments listed for the form.
- added the validate function to app-api
- added the submit function to app-api
- updated the package builder to update the package status based on the existence and placement of the RAI Withdraw event.
- updated package details to include the Withdraw RAI Response submission as part of the RAI Response section (for now)

### Implementation Notes
- the status "RAI Response Withdraw Enabled" is in the AC, but is dependent on the choices made for implementation of 24562.
- the changes to the details page are specifically to show the success of the form submission.  Please note, there is a story in the Backlog to reorganize the details page.

### Test Plan
1. Login as a submitting user
2. Navigate to dashboard
3. locate a package in "RAI Response Withdraw Enabled" status
4. click the Action dots, verify "Withdraw Formal RAI Response" is a potential action and that clicking the action takes you to the correct form.
5. Return to dashboard, click into the package in "RAI Response Withdraw Enabled" status and verify that "Withdraw Formal RAI Response" is in the available Package Actions links and that clicking the action takes you to the correct form.
6. Verify that only the Additional Information is needed to submit.
7. Verify that successfully submitting the package updates the package status to "Formal RAI Response - Withdrawal Requested"
8. Verify that the package details now shows the submitted RAI and the withdrawal as part of the RAI Responses section (NOTE: this is slightly different from the story AC and is only temporary, as there is a future story to reorganize the details page.)
